fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonPredicateConverter.java

```diff
@@ -171,9 +171,7 @@ public Predicate visitLikePredicateOperator(LikePredicateOperator operator, Void
         if (operator.getLikeType() == LikePredicateOperator.LikeType.LIKE) {
             if (operator.getChild(1).getType().isStringType()) {
                 String literal = ((BinaryString) getLiteral(operator.getChild(1))).toString();
-                if (literal != null && literal.length() > 1 &&
-                        literal.indexOf("%") == literal.length() - 1 &&
-                        literal.charAt(0) != '%') {
+                if (literal.length() > 1 && literal.indexOf("%") == literal.length() - 1 && literal.charAt(0) != '%') {
```

Comment:
Do we have any UT to cover this case?

Reply:
PaimonPredicateConverterTest.testBinaryString() covered.